### PR TITLE
Np 3598 populate scopus generator

### PR DESCRIPTION
**Pull request by axthosarouris**
Sorry for the big PR. It will a bit smaller once we merge the previous one. This serves as an example on how I believe the random data generator can help us with the conversion.

### CODE REVIEW

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 1]
```diff
@@ -114,15 +116,16 @@ void shouldLogExceptionMessageWhenExceptionOccurs() {
...
-        var scopusFile = IoUtils.stringFromResources(Path.of(SCOPUS_XML_0000469852));
-        var uri = s3Driver.insertFile(UnixPath.of(randomString()), scopusFile);
+    void shouldExtractOnlyScopusIdentifierIgnoreAllOtherIdentifiersAndStoreItInPublication() throws IOException {
+        var scopusData = new ScopusGenerator();
+        var scopusIdentifiers = keepOnlyTheScopusIdentifiers(scopusData);
+        var uri = s3Driver.insertFile(UnixPath.of(randomString()), scopusData.toXml());
         var s3Event = createS3Event(uri);
         var createPublicationRequest = scopusHandler.handleRequest(s3Event, CONTEXT);
         var actualAdditionalIdentifiers = createPublicationRequest.getAdditionalIdentifiers();
-        var expectedIdentifier =
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:123]: In the previous code, we have an inconsistency between the test and the code, which is actually visible in the test.
According to the code we may have more than one scopus identifiers for the same object and we will store them all. 
But here we are testing only the most common case, which is one identifier. 

By introducing the random data generator, we can generate the actual case and test that we will store all scopus identifiers. Note, that at this point we are not interested in whether expecting one or more scopus identifiers is correct or not. We are only interested in expressing what has been implemented.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/test/utils/ScopusGenerator.java [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    }
+
+    private static ShortTitle randomShortTitle() {
+        ShortTitle shortTitle = new ShortTitle();
+        shortTitle.setLang(randomScopusLanguageCode());
+        shortTitle.getContent().addAll(randomSerializables());
+        return shortTitle;
+    }
+
+    private static List<Serializable> randomSerializables() {
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/test/utils/ScopusGenerator.java:177]: this needs to be annotated and expanded to include multiple serializables.

File: create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java [hunk 3]
```diff
@@ -114,15 +116,16 @@ void shouldLogExceptionMessageWhenExceptionOccurs() {
...
-            new AdditionalIdentifier(ADDITIONAL_IDENTIFIERS_SCOPUS_ID_SOURCE_NAME, SCP_ID_IN_0000469852);
```
Comment (axthosarouris) [inline create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/ScopusHandlerTest.java:124]: removed hardcoded dependencies between resource files and values. If at some point we willl update the resources files, we do not have to update the test.

**Review by gregorgabriel — COMMENTED** [review]
I see now we you are aiming at. It is a very nice approach in generalizing the test/test cases. Thats defenitly a big plus.
On the other hand we will get a lot of code that has to be maintained (by gaining little, in my personal point of view). 
As this is a part of NVA code, the decision is up to you. Thats why I abstain my vote.

**Review by anetteOlli — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**